Lib/test/test_math.py

```diff
@@ -2503,6 +2503,50 @@ def test_input_exceptions(self):
         self.assertRaises(TypeError, math.atan2, 1.0)
         self.assertRaises(TypeError, math.atan2, 1.0, 2.0, 3.0)
 
+    def test_exception_messages(self):
+        x = -1.1
+
+        with self.assertRaises(ValueError,
+                               msg=f"expected a nonnegative input, got {x}"):
+            math.sqrt(x)
+
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {x}"):
+            math.log(x)
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {x}"):
+            math.log(123, x)
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {x}"):
+            math.log2(x)
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {x}"):
+            math.log2(x)
+
+        x = decimal.Decimal(x)
+
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {x!r}"):
+            math.log(x)
+
+        x = fractions.Fraction(1, 10**400)
+
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {float(x)!r}"):
```

Comment:
Can it be something except 0.0?

Reply:
Yes, above were tests for negative values.